app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :user
+  after_save :update_user_counter
+
+  def update_user_counter
+    user.update(PostsCounter: count)
+  end
```

Comment:
- [ ] The `count` variable you are using here is undefined, Please make sure you use valid code to update the counter. I would suggest you to using the below code.

```ruby
def update_user_counter
    user.increment!(:PostsCounter)
  end
```

- [ ] [Optional] I would suggest you make this method private So it can't be accessed outside of this class.